text/0436-gamelift-l2.md

```diff
@@ -0,0 +1,983 @@
+in the *Amazon GameLift Developer Guide*.
+
+##### Defining an instance type
+
+GameLift uses Amazon Elastic Compute Cloud (Amazon EC2) resources, called instances, to deploy your game servers and host game sessions for your players. When setting up a new fleet, you decide what type of instances your game needs and how to run game server processes on them (using a runtime configuration). All instances in a fleet use the same type of resources and the same runtime configuration. You can edit a fleet's runtime configuration and other fleet properties, but the type of resources cannot be changed.
+
+```ts fixture=with-build
+import * as gamelift from '@aws-cdk-lib/aws-gamelift';
+import * as ec2 from '@aws-cdk-lib/aws-ec2';
+
+new gamelift.Fleet(this, 'Game server fleet', {
+  build: build,
+  instanceType: ec2.InstanceType.of(ec2.InstanceClass.C5, ec2.InstanceSize.LARGE)
+});
+```
+
+##### Using Spot instances
+
+When setting up your hosting resources, you have the option of using Spot Instances, On-Demand Instances, or a combination.
+
+By default, this property is set to ON_DEMAND.
+
+```ts fixture=with-build
+import * as gamelift from '@aws-cdk-lib/aws-gamelift';
+
+new gamelift.Fleet(this, 'Game server fleet', {
+  build: build,
+  type: FleetType.SPOT
+});
+```
+
+##### Allowing Ingress traffic
+
+The allowed IP address ranges and port settings that allow inbound traffic to access game sessions on this fleet.
+
+New game sessions are assigned an IP address/port number combination, which must fall into the fleet's allowed ranges. Fleets with custom game builds must have permissions explicitly set. For Realtime Servers fleets, GameLift automatically opens two port ranges, one for TCP messaging and one for UDP.
+
+```ts fixture=with-build
+import * as gamelift from '@aws-cdk-lib/aws-gamelift';
+import * as ec2 from '@aws-cdk-lib/aws-ec2';
+
+const fleet = new gamelift.Fleet(this, 'Game server fleet', {
+  build: build,
+});
+// Allowing all IP Addresses from port 1111 to port 1122 on TCP Protocol
+fleet.addIngressRule(ec2.Peer.anyIpv4(), ec2.Port.tcp(1111), ec2.Port.tcp(1122));
+
+// Allowing a specific CIDR for port 1111 on UDP Protocol
+fleet.addIngressRule(ec2.Peer.ipv4('1.2.3.4/32'), ec2.Port.udp(1111));
+```
+
+##### Managing locations
+
+A single Amazon GameLift fleet has a home Region by default (the Region you deploy it to), but it can deploy resources to any number of GameLift supported Regions. Select Regions based on where your players are located and your latency needs.
+
+By default Stack region is used.
+
+```ts fixture=with-build
+import * as gamelift from '@aws-cdk-lib/aws-gamelift';
+
+new gamelift.Fleet(this, 'Game server fleet', {
+  build: build
+});
+```
+
+but we can add new locations using dedicated methods.
+
+```ts fixture=with-build
+import * as gamelift from '@aws-cdk-lib/aws-gamelift';
+
+const fleet = new gamelift.Fleet(this, 'Game server fleet', {
+  build: build
+});
+fleet.addLocation('eu-west-1');
```

Comment:
Please consider using the declarative style for all these other mutation methods too. Like `addLocation`, `addAlias`, `addIngressRule` and so on, as well as other constructs, like the one for runtime configuration.

Reply:
You mean adding possibility to declare it directly in the constructor?

Reply:
Yes.